li-utils/src/main/pegasus/com/linkedin/common/DashboardUrn.pdl

```diff
@@ -20,11 +20,11 @@ namespace com.linkedin.common
     "maxLength" : 20
   },
   {
-    "name" : "dashboardId",
-    "doc" : "Unique id of the dashboard assigned by the specific dashboard tool",
-    "type" : "string",
-    "maxLength" : 20
+    "name" : "dashboardUrl",
```

Comment:
Is the assumption that the dashboardUrl should be unique among dashboards so it is replacing the dashboardId? If so sounds good to me 👍

Reply:
Yes, that's the idea. Using URL itself is guaranteed to be unique even when there are multiple deployments of same dashboarding platform. Say you have 2 Looker deployments. Although UUIDs for dashboards are unique within the same deployment of Looker, it's not guarenteed to be unique globally. So, using the URL itself is a safer bet for global uniqueness. By this, we allow to have two dashboards with URLs `looker-1.spothero.com/dashboards/1` and `looker-2.spothero.com/dashboards/1`. Using that as a link to view the dashboard is also added benefit :)

Reply:
Can we still keep the original field name (i.e. `dashboardId`) but add comments that use the URL in the case of Looker?

Reply:
@mars-lan I don't think this is a problem specific to Looker. If there are multiple deployments of a dashboarding tool within a company, it's again the same problem (dashboard ids are unique locally but not globally). Therefore, highly likely, other dashboarding tools will want to use a URL scheme for this field. In that case, wouldn't it make sense to narrow down the scope of this field to reduce the confusion?

Reply:
Understand that, but instead of forcing people to always use URL here, why not be more generic and say "it's a unique ID, and if it URL works for you that's great"?

Reply:
+1. Agree with Mars on this. Might as well just call it `id` and let users select the right unique id for their use case.

Reply:
Sounds good, thanks for the comments @mars-lan @hshahoss . I will change the field names back to the original notation. Once we have this change, we can't rely on chart/dashboard URNs to extract URL from the URN. We need to store these external URLs explicitly in the aspects. Therefore, I will add fields to ChartInfo and DashboardInfo aspects to capture URLs for those.

Reply:
Addressed in the latest commit.